app/src/main/java/org/schabi/newpipe/database/Migrations.java

```diff
@@ -188,6 +189,15 @@ public void migrate(@NonNull final SupportSQLiteDatabase database) {
         }
     };
 
+    public static final Migration MIGRATION_5_6 = new Migration(DB_VER_5, DB_VER_6) {
+        @Override
+        public void migrate(@NonNull final SupportSQLiteDatabase database) {
+            database.execSQL("DELETE FROM search_history WHERE id NOT IN (SELECT id FROM "
+                    + "(SELECT id FROM search_history GROUP BY trim(search), service_id) tmp)");
```

Comment:
I don't know why SQL lets you do this, but you can't select one `id` from the `GROUP BY`, since there could be multiple `id`s in the same group, as you are not grouping by id. I would suggest to do `SELECT MIN(id)` for a more well-defined behavior, even though there would be no actual difference.

Reply:
Oops! It is my mistake due to oversight.